src/Shared/Model/Providers/NuGetProvider.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model.Providers;
+
+using Contracts;
+using Metadata;
+using Microsoft.CST.OpenSource.PackageManagers;
+using NLog;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using PackageUrl;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Utilities;
+
+public class NuGetProvider : IManagerProvider
+{
+    private readonly SourceCacheContext _sourceCacheContext = new();
+    private readonly SourceRepository _sourceRepository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+    private readonly NuGetLogger _logger = new(LogManager.GetCurrentClassLogger());
+
+    /// <summary>
+    /// Instantiates a new <see cref="NuGetProvider"/>.
+    /// </summary>
+    public NuGetProvider()
+    {
+    }
```

Comment:
Nit: this looks equivalent to the default constructor that C# automatically provides. Should we delete this in favor of the default constructor?